### PR TITLE
[ICDS // urgent] change the multiselect widget in locations to the select2 async 

### PR DESCRIPTION
**Pull request by biyeun**
@emord / @sravfeyn 

The current multiselect widget is not good for domains with a ton of users. 
https://manage.dimagi.com/default.asp?250165#1313044

### CODE REVIEW

File: corehq/apps/locations/forms.py [hunk 1]
```diff
@@ -1,6 +1,9 @@
+import json
```
Comment (dimagimon) [inline corehq/apps/locations/forms.py]: F401 'json' imported but unused

**Review by sravfeyn — APPROVED** [review]

File: corehq/apps/locations/forms.py [hunk 2]
```diff
@@ -333,38 +337,60 @@ def save(self, instance=None, commit=True):
...
 
     @property
     @memoized
     def users_at_location(self):
-        return (UserES()
-                .domain(self.domain_object.name)
-                .mobile_users()
-                .location(self.location.location_id)
-                .get_ids())
+        user_query = UserES().domain(
```
Comment (emord) [inline corehq/apps/locations/forms.py]: Formatting looks a little weird here. the convention is to use something similar to what it was before

```
(UserES()
 .domain(domain).
 .filter(...))
```

File: corehq/apps/locations/forms.py [hunk 3]
```diff
@@ -333,38 +337,60 @@ def save(self, instance=None, commit=True):
...
+            self.domain_object.name
+        ).mobile_users().location(
+            self.location.location_id
+        ).fields(['_id', 'username', 'first_name', 'last_name'])
+        return [
+            dict(id=u['_id'], text=user_display_string(
```
Comment (emord) [inline corehq/apps/locations/forms.py]: Might be easier to read as 
```
[
  {"id": u['_id'], "text": {user_display_string(...)}
  for u in query.run().hits
]
```

**Review by emord — APPROVED** [review]
Just some small nit picks. Tests failed on a network error. just bumped them

**Pull request merged**